tests/Unit/Domain/Test_DomainHelpers.cpp

```diff
@@ -0,0 +1,158 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Domain/Direction.hpp"
+#include "Domain/DomainHelpers.hpp"
+
+// Tests that get_common_global_corners is an order-sensitive intersection.
+TEST_CASE("Unit.Domain.DomainHelpers.GlobalCommon", "[Domain][Unit]") {
+  std::array<size_t, 8> block1{{0, 1, 2, 3, 4, 5, 6, 7}};
+  std::array<size_t, 8> block2{{9, 8, 7, 6, 5, 4, 3, 2}};
+  std::vector<size_t> common1{2, 3, 4, 5, 6, 7};
+  std::vector<size_t> common2{7, 6, 5, 4, 3, 2};
+  CHECK(DomainHelpers<3>::get_common_global_corners(block1, block2) == common1);
+  CHECK(DomainHelpers<3>::get_common_global_corners(block2, block1) == common2);
+}
+
+TEST_CASE("Unit.Domain.DomainHelpers.LocalCommon", "[Domain][Unit]") {
+  std::array<size_t, 8> block1_global{{10, 11, 12, 13, 14, 15, 16, 17}};
+  std::array<size_t, 4> block1_global_shared{{12, 14, 15, 10}};
+  std::array<size_t, 4> expected_local_common{{2, 4, 5, 0}};
+  CHECK(DomainHelpers<3>::get_common_local_corners(
+            block1_global, block1_global_shared) == expected_local_common);
+}
+
+TEST_CASE("Unit.Domain.DomainHelpers.LogicalCoordinates", "[Domain][Unit]") {
+  auto points1 = DomainHelpers<3>::logical_coords_of_corners({{0, 1, 2, 3}});
+  CHECK((points1[0] == Point<3, Frame::Logical>{{{-1.0, -1.0, -1.0}}}));
```

Comment:
OK.  Put them only around the right-hand side then so it doesn't mess up the catch output.